### PR TITLE
Adding unsupported to cluster validation

### PR DESCRIPTION
**Pull request by kris-nova**
@justinsb @chrislovecnm 

Can we try to get this in before the release?

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1158)

### CODE REVIEW

File: pkg/apis/kops/validation.go [hunk 1]
```diff
@@ -377,6 +378,42 @@ func (c *Cluster) Validate(strict bool) error {
...
+}
+
+type ClusterSpecBlackList struct {
+	KubernetesVersions []string
+}
+
+// Todo Kris - We need to get this into a yaml file that can be read in at compile time.. or do we?
+// If we decide to put this in a model of sorts, we will need to generate the bin data for it..
+// kind of sounds like we are walking down the model path again - so trying things out like this for now
+var CSBlacklist = ClusterSpecBlackList{KubernetesVersions: []string{"1.5.0", "v1.5.0"}}
```
Comment (kris-nova) [inline pkg/apis/kops/validation.go]: @chrislovecnm @justinsb do we want any other k8s versions blacklisted here?

Comment (justinsb) [inline pkg/apis/kops/validation.go]: Probably yes, but I think it's more important to get it into a dynamically loaded file

I don't know if we need YAML - a simple list of strings might suffice

File: pkg/apis/kops/validation.go [hunk 2]
```diff
@@ -377,6 +378,42 @@ func (c *Cluster) Validate(strict bool) error {
...
+
+
+// Will validate our cluster against the blacklist
+func (c *Cluster) isValidAgainstBlacklist() (error) {
+
+	isb := os.Getenv("KOPS_BYPASS_BLACKLIST")
+	if isb != "" {
```
Comment (justinsb) [inline pkg/apis/kops/validation.go]: Inline the variable, and glog.Warningf a message of what is happening

File: pkg/apis/kops/validation.go [hunk 3]
```diff
@@ -377,6 +378,42 @@ func (c *Cluster) Validate(strict bool) error {
...
+		// They are sure they want to bypass the validation check
+		return nil
+	}
+
+	// --------------------------------------------------------------------------------
+	// Kubernetes Version
+	//
+	for _, blacklistVersion := range CSBlacklist.KubernetesVersions {
+		if c.Spec.KubernetesVersion == blacklistVersion {
+			return fmt.Errorf("blacklisted kubernetes version %s detected. Bypass by setting $KOPS_BYPASS_BLACKLIST=1", blacklistVersion)
```
Comment (justinsb) [inline pkg/apis/kops/validation.go]: I think it would be `export KOPS_BYPASS_BLACKLIST=1`, right?  I'm not familiar with the `$` syntax?

Comment (chrislovecnm) [inline pkg/apis/kops/validation.go]: `$ KOPS_BYPASS_BLACKLIST=1` <- $ is the console.  You do not have to export it now.  The program will pick it up....

File: pkg/apis/kops/validation.go [hunk 4]
```diff
@@ -377,6 +378,42 @@ func (c *Cluster) Validate(strict bool) error {
...
+	// Blacklisted Values
+	err = c.isValidAgainstBlacklist()
+	if err != nil {
+		return fmt.Errorf("Unable to validate against blacklist: %v", err)
+	}
+
+	return nil
+}
+
+type ClusterSpecBlackList struct {
```
Comment (chrislovecnm) [inline pkg/apis/kops/validation.go]: `ClusterSpecK8sUnsupportList` update for happy happy joy joy

File: pkg/apis/kops/validation.go (hunk 1 above)
Comment (chrislovecnm) [inline pkg/apis/kops/validation.go]: Please make a map.  Version of k8s ... wth it is not supported.

**Comment by chrislovecnm** [comment]
@kris-nova we closing this??

**Comment by kris-nova** [comment]
Yep - going to rebuild this with a config file out of the binary

**Pull request closed without merging**